src/services/download/mod.rs

```diff
@@ -34,25 +36,30 @@ pub use crate::types::ObjectId;
 /// HTTP User-Agent string to use.
 const USER_AGENT: &str = concat!("symbolicator/", env!("CARGO_PKG_VERSION"));
 
-#[derive(Debug, Fail, Clone)]
-pub enum DownloadErrorKind {
-    #[fail(display = "failed to download")]
-    Io,
-    #[fail(display = "bad file destination")]
-    BadDestination,
-    #[fail(display = "failed writing the downloaded file")]
-    Write,
-    #[fail(display = "download was cancelled")]
+/// Errors happening while downloading from sources.
+#[derive(Debug, Error)]
+pub enum DownloadError {
+    #[error("Failed to download")]
+    Io(#[source] std::io::Error),
+    #[error("Failed to download stream")]
+    Stream(#[source] failure::Compat<PayloadError>),
+    #[error("Bad file destination")]
+    BadDestination(#[source] std::io::Error),
+    #[error("Failed writing the downloaded file")]
+    Write(#[source] std::io::Error),
+    #[error("Download was cancelled")]
     Canceled,
-    #[fail(display = "failed to fetch data from Sentry")]
-    Sentry,
+    #[error("Failed to fetch data from GCS")]
+    Gcs(#[from] gcs::GcsError),
+    #[error("Failed to fetch data from Sentry")]
+    Sentry(#[from] sentry::SentryError),
 }
 
-symbolic::common::derive_failure!(
-    DownloadError,
-    DownloadErrorKind,
-    doc = "Errors happening while downloading from sources."
-);
+impl DownloadError {
+    pub fn stream(err: PayloadError) -> Self {
```

Comment:
Not sure I like this naming much, reads more as a method than a constructor.

Reply:
probably `from_stream` would be better. The intention here was to make those method names as short as possible since we use them in a lot of places and I wanted to avoid ugly line breaking.